docs/resources/rules/assertions.md

```diff
@@ -0,0 +1,487 @@
+
+## Context object
+
+Property | Type | Description
+-- | -- | --
+type | `string` | **REQUIRED.** One of the [OpenAPI node types](#openapi-node-types).
+filterKeys | [`string`] | The list of key names to continue evaluation with respect to the subject.
+
+See the [context example](#context-example).
+
+## Sort Order object
+
+Property | Type | Description
+-- | -- | --
+direction | `string` | **REQUIRED.** Asserts values are sorted in ascending or descending order using one of these possible values: `asc` or `desc`.
+property | `string` | **REQUIRED.** The name of the property in the collection of objects that will hold values to assess the sort order.
+
+## Examples
+
+The following example shows two assertions each multiple asserts (`defined`, `minLength`, `maxLength`, `pattern`).
+
+The `Operation`, `Tag`, and `Info` properties must:
+- be defined
+- have at least 30 characters
+- end with a _full stop_.
+
+In addition, the `Operation` summary property must:
+- be defined
+- be between 20 and 60 characters
+- not end with a _full stop_.
+
+The following shows how to write that configuration:
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: Tag
+        property: description
+        message: Tag description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Operation
+        property: description
+        message: Operation description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Info
+        property: description
+        message: Info description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Operation
+        property: summary
+        message: Operation summary must be between 20 and 60 characters and not end with a full stop.
+        severity: error
+        defined: true
+        minLength: 20
+        maxLength: 60
+        pattern: /[^\.]$/
+```
+
+### `context` example
+
+The following example asserts that PUT responses with HTTP status 200 or 201 cannot return an application/pdf content type.
+Without the `context`, the assertion would evaluate every MediaTypeMap including:
+- Responses with all codes including others than 200 or 201
+- Responses for all HTTP methods including delete, get, post, and more.
+To restrict the evaluation, use the context feature to limit where context is used.
+
+```yaml
+- context:
+    - type: Operation
+      filterKeys: [put]
+    - type: ResponsesMap
+      filterKeys: [201, 200]
+  subject: MediaTypeMap
+  disallowed: ['application/pdf']
+```
```

Comment:
I like the idea of `context`, especially how it expands the previous "dot notation" way to traverse Node types. But I see there couple things to improve:

1. It's not clear from the notation above what is the Node types hierarchy - I understand that it's Operation -> ResponsesMap, but would be nice to introduce it clearer, like `next` key (or more suitable word) 
    ```js
    - context:
        - type: Operation
          filterKeys: [put]
          next:
          - type: ResponsesMap
            filterKeys: [201, 200]
    ```
    It will also expand the `context` power - now `context` will spread in many directions, like in the next example 
    ```js
    - context:
        - type: Operation
          filterKeys: [put]
          next:
          - type: ResponsesMap
            filterKeys: [201, 200]
        - type: Type1
          filterKeys: [key1, key2]
          next:
          - type: Type2
            filterKeys: [key3]
            next:
            - type: Type3
              filterKeys: [key4]     
    ```
    It tells us that the first context "direction" is Operation -> ResponsesMap, and the second Type1 -> Type2 -> Type3

2. I would rather use `onKeys` not `filterKeys`, as it sounds like "filter out". For me it's clearer :) 

3. It would be nice to have ability to exclude keys, with `offKeys` (or more suitable word) :
    ``` js
    - context:
      - type: Operation
        onKeys: [put]
        next:
        - type: ResponsesMap
          offKeys: [201, 200]
    ```

Reply:
@ohorbachevskyi thanks for the great feedback!

1. I believe you did not get the context correctly. The context is not an array of alternative contexts, this is actually a hierarchy. I see why you got confused though. But I think your suggestion is over-complicated and won't be used in any real use-cases. I would keep it as is.

2. I believe we were going to rename it to `filterParentKeys`. But I like `onParentKeys` much more. @adamaltman what do you think?

3. I'm not sure if `off` is the correct usage here but I agree it would be nice to have some keyword like that.

Reply:
How about `matchKeys` and `excludeKeys`?

Reply:
@adamaltman I think it's important to add `parent` word into naming... so `matchParentKeys`, `excludeParentKeys`?

Reply:
@adamaltman Yeah, `matchKeys` and `excludeKeys` much better! @RomanHotsiy why parent is so important here?

Reply:
I think it depends on how well people read and understand the docs. The issue is I only understand it personally as the parent. Let's take `put` for example... I think of it as belonging to the `PathItem` but it won't work as a `matchKey` of the `PathItem`. Instead, it is viewed as the parent of the `Operation`.

Reply:
I got it, thnx for clarifying.